### PR TITLE
Article feeder

### PR DESCRIPTION
**Pull request by rybmat**

### CODE REVIEW

**Comment by SebastianMarzjan** [comment]
LG. Please verify the failures though

File: service/indexer-base/article-feeder/build.gradle [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    globalDefaults {
+        app {
+            dropwizardCommandClasspathConfig("article-feeder", "indexer-base.yaml")
+        }
+    }
+    production {
+        resources {
+            cpus = 0.2
+            mem = 300
+            instances = 1
```
Comment (sqreek) [inline service/indexer-base/article-feeder/build.gradle]: is 1 instance is enough. 
I suggest 2 for fallback

File: service/indexer-base/article-feeder/src/main/java/com/wikia/services/indexerbase/ArticleFeeder.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    try {
+      String content = getContent(wid, pid);
+      this.publish(new JSONObject()
+                       .put("wiki_id", wid)
+                       .put("article_id", pid)
+                       .put("html", content)
+                       .toString());
+    } catch (ArticleNotFoundException e) {
+      LOGGER.warning(
```
Comment (sqreek) [inline service/indexer-base/article-feeder/src/main/java/com/wikia/services/indexerbase/ArticleFeeder.java]: Exception should be rethrown, or logged

File: service/indexer-base/article-feeder/src/main/java/com/wikia/services/indexerbase/ArticleFeeder.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          String.format("Could not fetch content of article: wiki_id: %d, page_id: %d", wid, pid));
+    }
+
+    return true;
+  }
+
+  private String getContent(Integer wid, Integer pid) throws ArticleNotFoundException {
+    String domain = this.domainCache.getDomainById(wid);
+
+    ParseActionOption query = MWApi.createBuilder().domain(domain)
```
Comment (sqreek) [inline service/indexer-base/article-feeder/src/main/java/com/wikia/services/indexerbase/ArticleFeeder.java]: please reformat this, one line per dot, for increase readability

File: service/indexer-base/article-feeder/src/main/java/com/wikia/services/indexerbase/ArticleFeeder.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        .parseAction().prop(ParsePropEnum.TEXT).pageId(pid);
+
+    return Optional.ofNullable(query.get())
+        .filter(result -> !Optional.ofNullable(result.getError()).isPresent())
+        .map(result -> result.getParse().getText().getText())
+        .orElseThrow(ArticleNotFoundException::new);
+  }
+
+  private class ArticleNotFoundException extends Exception {
```
Comment (sqreek) [inline service/indexer-base/article-feeder/src/main/java/com/wikia/services/indexerbase/ArticleFeeder.java]: Consider `public` this exception. This is way is valid, but if more classess use this exception, it should be not connected directly with only one of them

File: settings.gradle [hunk 5]
```diff
@@ -53,4 +53,5 @@ include 'service:example'
 include 'service:central-swagger-ui'
 include 'service:indexer-base:xwiki-feeder'//no tests
 include 'service:indexer-base:xwiki-worker'//no tests
+include 'service:indexer-base:article-feeder'
```
Comment (sqreek) [inline settings.gradle]: if U don't have tests add //no tests comment here, and if you add tests, remove it

**Comment by sqreek** [comment]
except lack of tests and minor comments LGTM

File: service/indexer-base/article-feeder/src/main/java/com/wikia/services/indexerbase/ArticleFeeder.java (hunk 2 above)
Comment (rybmat) [inline service/indexer-base/article-feeder/src/main/java/com/wikia/services/indexerbase/ArticleFeeder.java]: It is logged, so what is the issue?

**Pull request merged**